lib/pbio/drv/sound/sound_ev3.c

```diff
@@ -0,0 +1,96 @@
+// SPDX-License-Identifier: MIT
+// Copyright (c) 2025 The Pybricks Authors
+
+#include <pbdrv/config.h>
+
+#if PBDRV_CONFIG_SOUND_EV3
+
+#include <stdint.h>
+
+#include <pbdrv/gpio.h>
+
+#include <tiam1808/ehrpwm.h>
+#include <tiam1808/hw/soc_AM1808.h>
+#include <tiam1808/hw/hw_syscfg0_AM1808.h>
+#include <tiam1808/hw/hw_types.h>
+#include <tiam1808/psc.h>
+
+#include "../drv/gpio/gpio_ev3.h"
+
+// The division factor of 40 gives decent coverage of the frequency range
+// 64 Hz - 24 kHz that the other drivers support.
+// (This is limited by the 16-bit period counter.)
+#define PWM_TIMEBASE_FREQ       (SOC_EHRPWM_0_MODULE_FREQ / 40)
+
+// Audio amplifier enable
+static const pbdrv_gpio_t pin_sound_en = PBDRV_GPIO_EV3_PIN(13, 3, 0, 6, 15);
+// Audio output pin
+#define SYSCFG_PINMUX3_PINMUX3_7_4_GPIO0_0 0
+static const pbdrv_gpio_t pin_audio = PBDRV_GPIO_EV3_PIN(3, 7, 4, 0, 0);
+
+void pbdrv_sound_stop() {
+    // Turn speaker amplifier off
+    pbdrv_gpio_out_low(&pin_sound_en);
+    // Clean up counter
+    HWREGH(SOC_EHRPWM_0_REGS + EHRPWM_TBCTL) |= EHRPWM_TBCTL_CTRMODE_STOPFREEZE;
+    EHRPWMWriteTBCount(SOC_EHRPWM_0_REGS, 0);
+}
+
+void pbdrv_beep_start(uint32_t frequency, uint16_t sample_attenuator) {
+    // This hardware doesn't implement volume control
+    (void)sample_attenuator;
+
+    if (frequency < 64) {
+        frequency = 64;
+    }
+    if (frequency > 24000) {
+        frequency = 24000;
+    }
```

Comment:
In ev3dev our limits were 100 and 10000. It is probably fine to go a bit lower here, but not much point in going higher as I think we start losing volume much above that.